### PR TITLE
Stats: Total Subscribers Card

### LINKED ISSUE #23048: Total Subscribers Card
Full Scope: https://github.com/Automattic/wordpress-mobile/issues/47
Design: [IqhXWz3Iir7RMb5XH5gGfZ-fi-883%3A4006](https://href.li/?https://www.figma.com/file/IqhXWz3Iir7RMb5XH5gGfZ/?node-id=883%3A4006)

Create Total Subscribers Card

### Scope 

- Total Subscribers Card without navigation

### PR DESCRIPTION
**Pull request by staskus**
Fixes #23048

image

## Solution

1. Updated `subscribersList` to return both subscribers and subscribers totals since they come from the same endpoint
2. Added Total Subscribers card, reusing the same UI from Insights

Note: Insights Totals Subscribers card may show different numbers since it also adds publicize numbers. Fix is made here https://github.com/wordpress-mobile/WordPress-iOS/pull/23099

## To test:

1. Enable Stats Traffic and Subscribers tab feature flag
2. Open Stats -> Subscribers
3. Confirm Total Subscribers card appears
4. Compare the number with the web

## Regression Notes
1. Potential unintended areas of impact

None

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

Added unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/286">24.8</a>. This milestone is due in less than 4 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/ViewRelated/Stats/Subscribers/StatsSubscribersStore.swift [hunk 1]
```diff
@@ -106,17 +107,20 @@ struct StatsSubscribersStore: StatsSubscribersStoreProtocol {
                 completion(.failure(error))
             } else {
                 // Combine both wpcom and email subscribers into a single list
-                let followers = Array(
-                    followers
-                        .sorted(by: { $0.subscribedDate > $1.subscribedDate })
-                        .prefix(quantity)
-                )
-                completion(.success(followers))
+                let combinedSubscribers = Array((wpComFollowers?.topDotComFollowers ?? []) + (emailFollowers?.topEmailFollowers ?? [])
```
Comment (staskus) [inline WordPress/Classes/ViewRelated/Stats/Subscribers/StatsSubscribersStore.swift:141]: Based on the answer here pejTkB-1l5-p2#comment-1316 the idea is for all of this aggregation to be replaced by `v2/sites/{site_id}/stats/subscribers`. I think we can do that both in scope and outside the scope of the project depending on time constraints.

**Review by guarani — APPROVED** [review]
Works great, thanks @staskus!

**Comment by aditi-bhatia** [comment]
Hi @staskus @guarani, I just wanted to double check something since I'm working on this for Android. Are we planning to implement the `% change` detail at the bottom of this card during this iteration? I'm not sure if I missed a conversation about this somewhere but I noticed it was in the designs: 

Screenshot 2024-04-29 at 10 56 28 PM

**Comment by staskus** [comment]
> I just wanted to double check something since I'm working on this for Android. Are we planning to implement the % change detail at the bottom of this card during this iteration? I'm not sure if I missed a conversation about this somewhere but I noticed it was in the designs

Good question, @aditi-bhatia. No, at least not in this iteration. 👍

**Comment by guarani** [comment]
> I'm not sure if I missed a conversation about this somewhere but I noticed it was in the designs

I couldn't see the comparison label in the designs mentioned in the issue https://github.com/wordpress-mobile/WordPress-iOS/issues/23048. I'm just asking to double-check that we're looking at the same designs.

**Comment by aditi-bhatia** [comment]
Ah good catch 🤔 I had been referencing this Figma IqhXWz3Iir7RMb5XH5gGfZ-fi-97_5069 from this discussion p1713440407948299-slack-C06BR07TJHK which looks like was a previous iteration. Thanks both!

**Pull request merged**